src/core/namespace.c

```diff
@@ -2286,13 +2287,27 @@ int setup_tmp_dirs(const char *id, char **tmp_dir, char **var_tmp_dir) {
         return 0;
 }
 
-int setup_netns(const int netns_storage_socket[static 2]) {
-        _cleanup_close_ int netns = -1;
+static const char *ns_filename(unsigned long nstype) {
```

Comment:
Please move this to `shared/nsflags.c` and maybe it is better to name this `namespace_type_to_string()`.

Reply:
The problem is we have a `namespace_type_to_string()` in `namespace.h` (it accepts a `NamespaceType` enum).  I'd like to rename `nstype` to `nsflag`, and use `namespace_single_flag_to_string` for it.  (`single` is explicitly added, to distinguish from `namespace_flags_to_string` better.)

Reply:
Oh, indeed. Then, `namespace_single_flag_to_string()` is ok. (Though, `_single` sounds slightly redundant for me, but not a strong opinion.)